quixstreams/sinks/community/iceberg.py

```diff
@@ -0,0 +1,203 @@
+            # Create the Iceberg table if it doesn't exist.
+            self.table = self.catalog.create_table_if_not_exists(
+                identifier=table_name,
+                schema=schema,
+                location=aws_s3_uri,
+                partition_spec=partition_spec,
+                properties={"write.distribution-mode": "fanout"}
+            )
+            self._logger.info(f"Loaded Iceberg table '{table_name}' at '{aws_s3_uri}'.")
+        
+    def write(self, batch: SinkBatch):
+        """
+        Writes a batch of data to the Iceberg table.
+        Implements retry logic to handle concurrent write conflicts.
+
+        Parameters:
+            batch (SinkBatch): The batch of data to write.
+        """
+        try:
+            # Serialize batch data into Parquet format.
+            data = self._format.serialize_batch_values(batch)
+            
+            # Read data into a PyArrow Table.
+            input_buffer = pa.BufferReader(data)
+            parquet_table = pq.read_table(input_buffer)
+
+            # Reload the table to get the latest metadata
+            self.table = self.catalog.load_table(self.table.name())
+            
+            # Update the table schema if necessary.
+            with self.table.update_schema() as update:
+                update.union_by_name(parquet_table.schema)
+            
+            append_start_epoch = time.time()
+            self.table.append(parquet_table)
+            self._logger.info(f"Appended {len(list(batch))} records to {self.table.name()} table in {time.time() - append_start_epoch}s.")
+            
+        except CommitFailedException as e:
+            # Handle commit conflict
+            self._logger.warning(f"Commit conflict detected.: {e}")
+            sleep_time = random.uniform(0, 5)
+            raise SinkBackpressureError(sleep_time, batch.topic, batch.partition)
+        except Exception as e:
+            self._logger.error(f"Error writing data to Iceberg table: {e}")
+            raise
+        
+    
+    def _serialize_batch_values(self, values: List[Any]) -> bytes:
+        #TODO: Handle data flattening. Nested properties will cause this to crash.
+        
+        # Get all unique keys (columns) across all rows
+        all_keys = set()
+        for row in values:
+            all_keys.update(row.value.keys())
+
+        # Normalize rows: Ensure all rows have the same keys, filling missing ones with None
+        normalized_values = [
+            {key: row.value.get(key, None) for key in all_keys} for row in values
+        ]
+
+        columns = {
+            "_timestamp": [datetime.fromtimestamp(row.timestamp / 1000.0) for row in values], 
+            "_key": [bytes.decode(row.key) for row in values]
+        }
+
+        # Convert normalized values to a pyarrow Table
+        columns = {**columns, **{key: [row[key] for row in normalized_values] for key in all_keys}}
+                   
+        table = pa.Table.from_pydict(columns)
```

Comment:
Just FYI: Pycharm claimed a missing parameter argument for `from_pydict()`, but at a glance your use looked correct to me based on the docs.